lark/visitors.py

```diff
@@ -194,18 +195,18 @@ class Transformer_NonRecursive(Transformer):
     Useful for huge trees.
     """
 
-    def transform(self, tree):
+    def transform(self, tree: Tree[_Leaf_T]) -> _Return_T:
         # Tree to postfix
-        rev_postfix = []
-        q = [tree]
+        rev_postfix: List[Union[_Leaf_T, Tree[_Leaf_T]]] = []
```

Comment:
Is this necessary? Why not just annotate `result`?

Reply:
This specific annotation isn't needed by `mypy`. I think I added it at the start of updating this method to help me keep track of what is happening with the various lists.

Based on the conversation with @MegaIng on the result line, I think I'll need to take a second pass at this whole file now that I have a better understanding.